logic/src/commonMain/kotlin/com/wire/kalium/logic/data/asset/AssetRepository.kt

```diff
@@ -207,7 +212,8 @@ internal class AssetDataSource(
                 assetApi.downloadAsset(assetId, assetToken?.ifEmpty { null }, tempFileSink)
             }.flatMap {
                 try {
-                    val encryptedAssetDataSource = kaliumFileSystem.source(tempFile)
+                    if (encryptionKey != null && assetSHA256 == null) return@flatMap Either.Left(EncryptionFailure.WrongAssetHash)
+                    val encryptionKeys = encryptionKey?.let { Pair(it, assetSHA256!!) }
```

Comment:
```suggestion
                    val encryptionKeys = encryptionKey?.let {
                        if (assetSHA256 == null) {
                            return@flatMap Either.Left(EncryptionFailure.WrongAssetHash)
                        } else {
                            Pair(it, assetSHA256)
                        }
                    }
```

Let's also get rid of this force unwrap.